anvio/cazymes.py

```diff
@@ -0,0 +1,259 @@
+            Filtering option for HMM search
+        - hmm_program : str, optional
+            hmmsearch (default) or hmmscan
+    run : terminal.Run, optional
+        An object for printing messages to the console.
+    progress : terminal.Progress, optional
+        An object for printing progress bars to the console.
+    """
+    def __init__(self, args, run=run, progress=progress):
+
+        self.run = run
+        self.progress = progress
+
+        A = lambda x, t: t(args.__dict__[x]) if x in args.__dict__ else None
+        null = lambda x: x
+        self.contigs_db_path = A('contigs_db', null)
+        self.num_threads = A('num_threads', null)
+        self.hmm_program = A('hmmer_program', null) or 'hmmsearch'
+        self.noise_cutoff_terms = A('noise_cutoff_terms', null)
+        self.cazyme_data_dir = args.cazyme_data_dir
+
+        # load_catalog will populate this
+        self.function_catalog = {}
+
+        filesnpaths.is_program_exists(self.hmm_program)
+        utils.is_contigs_db(self.contigs_db_path)
+
+        if not self.cazyme_data_dir:
+            self.cazyme_data_dir = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/CAZyme')
+
+        self.is_database_exists()
+
+        # reminder to be a good citizen (Good idea Iva!)
+        self.run.warning("Anvi'o will annotate your contigs-db genes with the dbCAN CAZyme database. "
+                         "Please do not forget to cite this database when you publish your science :) "
+                         "(http://www.cazy.org/Citing-CAZy).", lc='green', header="CITATION")
+
+    def is_database_exists(self):
+        """Checks if decompressed database files exist"""
+
+        if not glob.glob(os.path.join(self.cazyme_data_dir, "dbCAN-HMMdb-*.txt")):
+            raise ConfigError(f"It seems you do not have the CAZyme database installed in {self.cazyme_data_dir}, "
+                              f"please run 'anvi-setup-cazymes' download it.")
+
+        # Glob and find what files we have then check if we have them all
+        downloaded_files = glob.glob(os.path.join(self.cazyme_data_dir, '*'))
+        
+        # here we check if the HMM profile is compressed so we can decompress it for next time
+        hmmpress_file_extensions = ["h3f", "h3i", "h3m", "h3p", "txt"]
+        extant_extensions = [os.path.basename(file).split(".")[-1] for file in downloaded_files]
+
+        if hmmpress_file_extensions.sort() != extant_extensions.sort():
+            raise ConfigError("Anvi'o detected that the CAZyme database was not properly compressed with hmmpress. "
+                              "Please 'anvi-setup-cazymes --reset'")
+
+    def process(self):
+        """Search CAZyme HMMs over contigs-db, parse, and filter results"""
+
+        #FIXME: need a smarter way to find the CAZyme HMM file for when users 
+        # want a specific version. This will break if it's not V11
+        hmm_file = os.path.join(self.cazyme_data_dir, "dbCAN-HMMdb-V11.txt")
```

Comment:
Idea: you could rename the HMM file when you download it so that the name is always the same and not version-dependent. 

The only problem with this is that version information is then lost. Which you could potentially solve by adding a `version.txt` file within the CAZymes data folder so it can be looked up as needed.

Reply:
Great idea, just implemented :)